### ISSUE TITLE
Request to add support for Tweedie Distribution

### ISSUE THREAD

**Issue by Willamette-OR** [dmlc/xgboost#1392]
Tweedie distribution (w/ a log link of course to ensure non-negativity) is essential for the development of pricing models in the insurance industry. H2O has been supporting Tweedie & Gamma in their GBM implementation and therefore gained a lot of popularity among quite a few large insurance companies. Gamma is another popular distribution in solving insurance problems.

Would it be possible for someone to bring in support for Tweedie for XGBOOST? I've noticed Gamma distribution has recently been made available which is fantastic!

**Comment by jasdumas** [comment]
I second this issue request. I would find it very useful in the R package

**Comment by tqchen** [comment]
I think this should not be hard with the customized loss function. maybe @yangs16 (contributor of the gamma regression) have some more to say.

**Comment by marugari** [comment]
I understand that discussion above assumes using gblinear, however, we can use boosted trees for estimating Tweedie distribution.
http://arxiv.org/abs/1508.06378
I wish someone could add TDboost.

**Comment by khotilov** [comment]
@marugari : the discussion is about a general objective for Tweedie loss applicable to either booster. 

I've started looking into it a while ago, but didn't ever have time to fully go through. Besides, there seem to exist some subtleties about the estimation of Tweedie distribution parameters (another question would be though whether such estimation is really necessary in this specific context). The gbm package has Tweedie loss too, but there was something not quite right about it last time I was looking.

PS:
@Willamette-OR before posting a new issue, it's always a good idea to use the search box at the top to see whether a similar issue was already created before #681

**Comment by Willamette-OR** [comment]
@tqchen Thanks Tianqi. I was indeed aware of the customized loss function feature, but I also learned that xgb's parallelized calculation would be unavailable whenever I use this feature, which makes it less appealing when dealing with large datasets.

**Comment by yangs16** [comment]
As @khotilov said, Tweedie is somehow more complicated. I have no plan to add the support in near future. But I agree with @Willamette-OR that Tweedie should be a very useful feature for insurance pricing models.

**Comment by tqchen** [comment]
@Willamette-OR The parallelized tree construction is still available on the customized loss. While the loss function calculation is not parallelized, it is not the bottleneck so in general you can find customized loss as fast as the build in one
